packages/components/src/components/contract-card/contract-card-items/accumulator-card-body.tsx

```diff
@@ -1,13 +1,54 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
 import { isCryptocurrency, getLimitOrderAmount, isValidToSell } from '@deriv/shared';
 import ContractCardItem from './contract-card-item.jsx';
 import ToggleCardDialog from './toggle-card-dialog.jsx';
-import Icon from '../../icon';
-import MobileWrapper from '../../mobile-wrapper';
-import Money from '../../money';
+import Icon from '../../icon/index';
+import MobileWrapper from '../../mobile-wrapper/index';
+import Money from '../../money/index';
 import { ResultStatusIcon } from '../result-overlay/result-overlay.jsx';
+import { ProposalOpenContract, Proposal, ContractUpdate } from '@deriv/api-types';
+import { TIsValidToSell } from '@deriv/shared/src/utils/contract/contract-types';
+
+type TToastConfig = {
+    key?: number;
+    content: string;
+    timeout?: number;
+    is_bottom?: boolean;
+    type?: string;
+};
+
+type TConnectWithContractUpdateReturn = {
+    validation_errors: { contract_update_stop_loss: string[]; contract_update_take_profit: string[] };
+    contract_update_take_profit: number | string;
+    contract_update_stop_loss: number | string;
+    has_contract_update_take_profit: boolean;
+    has_contract_update_stop_loss: boolean;
+};
+
+export type TContractInfo = Pick<ProposalOpenContract, 'buy_price' | 'sell_price' | 'contract_id'> &
+    Required<Pick<ProposalOpenContract, 'profit'>> & {
+        limit_order?: Proposal['limit_order'];
+    } & TIsValidToSell;
+
+type TAccumulatorCardBody = {
+    addToast?: (toast_config: TToastConfig) => void;
+    connectWithContractUpdate?: (Component: JSX.Element) => TConnectWithContractUpdateReturn;
```

Comment:
```suggestion
    connectWithContractUpdate?: (Component: JSX.Element) => JSX.Element;
```
or
```
    connectWithContractUpdate?: (Component: React.ComponentType) => React.ComponentType;
```
Will this work?🙂 It looks like this callback accepts a components and returns a component

Reply:
returns this
```
export const connectWithContractUpdate = connect(({}, { contract = {} }) => {
    return {
        validation_errors: contract.validation_errors,
        contract_update_take_profit: contract.contract_update_take_profit,
        contract_update_stop_loss: contract.contract_update_stop_loss,
        has_contract_update_take_profit: contract.has_contract_update_take_profit,
        has_contract_update_stop_loss: contract.has_contract_update_stop_loss,
    };
});
```

Reply:
@henry-deriv yes, it's assigned with a return value of the connect() call, and connect returns a component (connect is a wrapper for a component which connects it to values from the store). Thus, in the end, connectWithContractUpdate returns a component connected to the data in the object above.
To see it's usage, you can refer to the line 44 in toggle-card-dialog.jsx:
`ContractUpdateFormWrapper = connectWithContractUpdate?.(ContractUpdateForm) || ContractUpdateForm;`
As we can see, connectWithContractUpdate() accepts a component (ContractUpdateForm) and returns a component (ContractUpdateFormWrapper)😊